### PR TITLE
Triage Improvements

### PR DESCRIPTION
**Pull request by rmmh**
- sparklines
- recent failures
- github search for cluster

Staging url: https://storage.googleapis.com/k8s-gubernator/triage/staging/index.html

### CODE REVIEW

File: triage/render.js [hunk 1]
```diff
@@ -63,43 +63,53 @@ function addElement(parent, type, opts, children) {
   return el;
 }
 
+// Creates a urlsafe slug out of a string-- MUST match Gubernator's slugify function.
```
Comment (fejta) [inline triage/render.js]: What is a slug?

File: triage/render.js [hunk 2]
```diff
@@ -113,10 +123,44 @@ function renderJobs(parent, buildsIterator) {
   var jobList = addElement(parent, 'ul');
   for (let [job, buildNumbersSet] of clusterJobs) {
     let buildNumbers = Array.from(buildNumbersSet).sort();
-    addBuildListItem(jobList, job, buildNumbers);
+    addBuildListItem(jobList, job, buildNumbers, counts[job]);
   }
 }
 
+function sparkLinePath(arr, width, height) {
```
Comment (fejta) [inline triage/render.js]: Please document this function. Also link to something that describes the SVG format and whatever a path is.

This is super cool but we cannot assume that your average JS engineer understands SVG

File: triage/render.js [hunk 3]
```diff
@@ -128,66 +172,75 @@ function renderCluster(top, key, keyId, text, tests) {
...
 
+  var counts = clustered.makeCounts(keyId);
+
   var clusterSum = clustersSum(tests);
   var recentCount = clustered.getHitsInLastDayById(keyId);
   var failureNode = addElement(top, 'div', {id: keyId}, [
-    createElement('h2',
-      {innerHTML: `${plural(clusterSum, 'FAILURE', 'S')} (${recentCount} RECENT) MATCHING <a href="#${keyId}" class="key">${keyId}</a>`}),
+    createElement('h2', null, [
+      `${plural(clusterSum, 'FAILURE', 'S')} (${recentCount} RECENT) MATCHING `,
```
Comment (fejta) [inline triage/render.js]: I do not understand what recent means. Can we change this to today? Sounds more clear.

File: triage/render.js [hunk 4]
```diff
@@ -128,66 +172,75 @@ function renderCluster(top, key, keyId, text, tests) {
...
+      createElement('a', {href: 'https://github.com/search?type=Issues&q=' + keyId, target: '_blank', rel: 'noopener'}, 'github search')
+    ]),
     createElement('pre', null, options.showNormalize ? key : text),
     createElement('div', {className: 'graph', dataset: {cluster: keyId}}),
   ]);
-  var clusterJobs = addElement(failureNode, 'ul');
-  var list = addElement(failureNode, 'ul');
+  var list = addElement(failureNode, 'ul', null, [
+    createElement('li', {innerText: `Recent Failures ${rightArrow}`}, [
+      createElement('div', {className: 'recent', dataset: {cluster: keyId}})
```
Comment (fejta) [inline triage/render.js]: Can we please just show the N most recent failures? Aka just make this always show and stick this inside an expanding details

File: triage/render.js [hunk 5]
```diff
@@ -128,66 +172,75 @@ function renderCluster(top, key, keyId, text, tests) {
...
+      createElement('a', {href: '#' + keyId}, keyId),
+      createElement('a', {href: 'https://github.com/search?type=Issues&q=' + keyId, target: '_blank', rel: 'noopener'}, 'github search')
+    ]),
     createElement('pre', null, options.showNormalize ? key : text),
     createElement('div', {className: 'graph', dataset: {cluster: keyId}}),
   ]);
-  var clusterJobs = addElement(failureNode, 'ul');
-  var list = addElement(failureNode, 'ul');
+  var list = addElement(failureNode, 'ul', null, [
+    createElement('li', {innerText: `Recent Failures ${rightArrow}`}, [
```
Comment (fejta) [inline triage/render.js]: Please make this a superlative so it is more clear (newest failures, most recent failures).

Also try and use a different word. Why does the H2 say 1316 RECENT FAILURES and yet when I expand recent failures there are only 5.

File: triage/render.js [hunk 6]
```diff
@@ -128,66 +172,75 @@ function renderCluster(top, key, keyId, text, tests) {
...
+
   var clusterSum = clustersSum(tests);
   var recentCount = clustered.getHitsInLastDayById(keyId);
   var failureNode = addElement(top, 'div', {id: keyId}, [
-    createElement('h2',
-      {innerHTML: `${plural(clusterSum, 'FAILURE', 'S')} (${recentCount} RECENT) MATCHING <a href="#${keyId}" class="key">${keyId}</a>`}),
+    createElement('h2', null, [
+      `${plural(clusterSum, 'FAILURE', 'S')} (${recentCount} RECENT) MATCHING `,
+      createElement('a', {href: '#' + keyId}, keyId),
+      createElement('a', {href: 'https://github.com/search?type=Issues&q=' + keyId, target: '_blank', rel: 'noopener'}, 'github search')
```
Comment (fejta) [inline triage/render.js]: Amazing! :clap: 

Would also love to see something I can click to file and issue that prepopulates the keyId in the body for me

File: triage/render.js [hunk 7]
```diff
@@ -113,10 +123,44 @@ function renderJobs(parent, buildsIterator) {
...
 
+function sparkLinePath(arr, width, height) {
+  var max = 0;
+  for (var i = 0; i < arr.length; i++) {
+    if (arr[i] > max)
+      max = arr[i];
+  }
+  var scale = max > 0 ? height / max : 1;
+
+  var out = 'M0,' + height;
```
Comment (fejta) [inline triage/render.js]: Will you also comment what all these string pieces are and their function?

**Comment by fejta** [comment]
Very awesome and amazing!!

LGTM modulo comments. Would like the fancy SVG stuff in this file to be over documented since this is not a typical skill for our team (I have created exactly 0 svg files).

Would like @cjwagner to apply the label

File: triage/render.js (hunk 1 above)
Comment (rmmh) [inline triage/render.js]: https://en.wikipedia.org/wiki/Semantic_URL#Slug

File: triage/render.js (hunk 2 above)
Comment (rmmh) [inline triage/render.js]: done

File: triage/render.js (hunk 3 above)
Comment (rmmh) [inline triage/render.js]: done

File: triage/render.js (hunk 5 above)
Comment (rmmh) [inline triage/render.js]: done

File: triage/render.js (hunk 7 above)
Comment (rmmh) [inline triage/render.js]: done

**Comment by fejta** [comment]
LGTM

**Comment by fejta** [comment]
Would still prefer Cole to weigh in

**Pull request merged**